File: Business/MergedEntity/Payroll.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DataAccess;

namespace Business.MergedEntity
{
    public class Payroll
    {
        public string id { get; set; }
        public string name { get; set; }
        public sbyte payroll_type { get; set; }
        public sbyte period_type { get; set; }
        public bool require_educate { get; set; }
        public double require_wage { get; set; }
        // public DateTime payroll_period { get; set; }
        public DateTime payroll_period { get; set; }
        public int standard_work_time { get; set; }
        public int real_work_day { get; set; } // Số ngày làm việc thực tế
        public int extra_weekday { get; set; } // Số giờ làm thêm vào ngày thường
        public int extra_weekend { get; set; } // Số giờ làm thêm vào ngày cuối tuần
        public int extra_holiday { get; set; } // Số giờ làm thêm vào ngày lễ tết
        public int completed_product_price { get; set; } // Tổng giá trị số sản phẩm hoàn thành
        public int restrict_product { get; set; } // Số sản phẩm khoán
        public int completed_restrict_product { get; set; }
        // public double price_per_product { get; set; } // Đơn giá một sản phẩm
        public double total { get; set; }
        public int absences { get; set; }
        public int normalExtraHours { get; set; }
        public int weekendExtraHours { get; set; }
        public int holidayExtraHours { get; set; }
        public List<Product> listProducts { get; set; }
    }
}

File: HRM/Scripts/Employee/payroll.js
$("#btn-calculate-payroll").on("click", (e) => {
    let employeeId;
    var completeEmployeeName;
    $(".select-employee .dropdown-item").each((idx, itm) => {
        if ($(itm).find("input").is(":checked") == true) {
            employeeId = $(itm).find("input").attr("id").split('-')[1].trim();
            completeEmployeeName = $(itm).find("label").text().trim();
        }
    });
    let payrollPeriod = $("#calendar-payroll-period").val();
    let prePayrollPeriod = $("#calendar-pre-payroll-period").val();

    $.ajax({
        url: '/Employee/GetPayrollInformation',
        contentType: 'application/x-www-form-urlencoded',
        dataType: 'json',
        data: { employeeId: employeeId, payrollPeriod: payrollPeriod, prePayrollPeriod: prePayrollPeriod },
        type: 'post',
        success: function (response) {
            let getCodeTool = GetValueOfCode();
            if (response) {
                // alert("Xóa nhân viên thành công!");
                // $("#modal-modify-employee").modal("toggle");
                // newEmployee.id = response.new_id;
                // window.allEmps = [newEmployee].concat(window.allEmps);
                // renderEmployees(window.allEmps);
                let educateExtra = 0;
                if (response.require_educate == true) {
                    educateExtra = 0.07 * response.require_wage;
                }

                let productHtml = '';
                $.each(response.listProducts, (idx, itm) => {
                    productHtml += `
                        <tr class="quantity-${itm.id} hide by-product">
                            <td>Sản phẩm ${itm.name}</td>
                            <td><span class="quantity">${itm.quantity}</span> * <span class="price">${itm.value}</span> = <span class="result">${itm.value * itm.quantity}</span></td>
                        </tr>
                    `;
                })

                $(".table-payroll").find("table").remove();
                $(".table-payroll").append(`
                    <table class="table table-bordered">
                        <thead>
                            <tr class="employee-info">
                                <th colspan="2">${completeEmployeeName}</th>
                            </tr>
                        </thead>
                        <tbody>
                            <tr class="payroll-method">
                                <td>Hình thức trả lương</td>
                                <td>${getCodeTool.getPayrollMethod(response.payroll_type)}</td>
                            </tr>
                            <tr class="payroll-period">
                                <td>Kì trả lương</td>
                                <td>${getCodeTool.getPayrollPeriod(response.period_type)}</td>
                            </tr>
                            <tr class="require-way">
                                <td>Lương thỏa thuận</td>
                                <td>${response.require_wage}</td>
                            </tr>
                            <tr class="educate-bonus">
                                <td>Phụ cấp đào tạo</td>
                                <td>${educateExtra}</td>
                            </tr>
                            <tr class="area-wage">
                                <td>Lương vùng</td>
                                <td>4420000</td>
                            </tr>
                            <tr class="not-allowed-absence hide by-time">
                                <td>Nghỉ không phép</td>
                                <td>${response.absences}</td>
                            </tr>
                            <tr class="normal-extra hide by-time">
                                <td>Làm thêm vào ngày thường</td>
                                <td>${response.normalExtraHours}</td>
                            </tr>
                            <tr class="weekend-extra hide by-time">
                                <td>Làm thêm vào ngày cuối tuần</td>
                                <td>${response.weekendExtraHours}</td>
                            </tr>
                            <tr class="holiday-extra hide by-time">
                                <td>Làm thêm vào ngày lễ tết</td>
                                <td>${response.holidayExtraHours}</td>
                            </tr>
                            ${productHtml}
                            <tr class="restrict hide by-restrict">
                                <td>Khoán</td>
                                <td>${response.restrict_product}</td>
                            </tr>
                            <tr class="restrict-complete hide by-restrict">
                                <td>Sản phẩm khoán hoàn thành</td>
                                <td>${response.completed_restrict_product}</td>
                            </tr>
                            <tr class="total-payroll">
                                <td>Thực nhận</td>
                                <td>${response.total}</td>
                            </tr>
                        </tbody>
                    </table>
                `);

                if (response.payroll_type == 1) {
                    $(".by-time").removeClass("hide");
                }
                else if (response.payroll_type == 2) {
                    $(".by-product").removeClass("hide");
                }
                else {
                    $(".by-restrict").removeClass("hide");
                }

                // location.reload();
            }
        }
    });
})